t/25-cache-service.t

```diff
@@ -0,0 +1,454 @@
+
+    $_->start for ($worker_2, $worker_3, $worker_4);
+
+    my @assets = map { "sle-12-SP3-x86_64-0368-200_$_\@64bit.qcow2" } 1 .. $tot_proc;
+    unlink path($cachedir)->child($_) for @assets;
+    ok(
+        $cache_client->enqueue(
+            OpenQA::Worker::Cache::Request->new(client => $cache_client)
+              ->asset(id => 922756, asset => $_, type => 'hdd', host => $host)
+        ),
+        "Download enqueued for $_"
+    ) for @assets;
+
+    sleep 1
+      until (
+        grep { $_ == 1 } map {
+            $cache_client->processed(OpenQA::Worker::Cache::Request->new(client => $cache_client)
+                  ->asset(id => 922756, asset => $_, type => 'hdd', host => $host))
+        } @assets
+      ) == @assets;
+
+    ok($cache_client->asset_exists('localhost', $_), "Asset $_ downloaded correctly") for @assets;
+
+    @assets = map { "sle-12-SP3-x86_64-0368-200_88888\@64bit.qcow2" } 1 .. $tot_proc;
+    unlink path($cachedir)->child($_) for @assets;
+    ok(
+        $cache_client->enqueue(
+            OpenQA::Worker::Cache::Request->new(client => $cache_client)
+              ->asset(id => 922756, asset => $_, type => 'hdd', host => $host)
+        ),
+        "Download enqueued for $_"
+    ) for @assets;
+
+    sleep 1
+      until (
+        grep { $_ == 1 } map {
+            $cache_client->processed(OpenQA::Worker::Cache::Request->new(client => $cache_client)
+                  ->asset(id => 922756, asset => $_, type => 'hdd', host => $host))
+        } @assets
+      ) == @assets;
+
+    ok($cache_client->asset_exists('localhost', "sle-12-SP3-x86_64-0368-200_88888\@64bit.qcow2"),
+        "Asset $_ downloaded correctly")
+      for @assets;
+
+    $_->stop for ($worker_2, $worker_3, $worker_4);
+};
+
+subtest 'Test Minion task registration and execution' => sub {
+    my $a = 'sle-12-SP3-x86_64-0368-200_133333@64bit.qcow2';
+
+    use Minion;
+    my $minion = Minion->new(SQLite => "sqlite:" . OpenQA::Worker::Cache->from_worker->db_file);
+    use Mojolicious;
+    my $app = Mojolicious->new;
+    $app->attr(minion => sub { $minion });
+
+    my $task = OpenQA::Worker::Cache::Task::Asset->new();
+    $task->register($app);
+    my $req = OpenQA::Worker::Cache::Request->new(client => $cache_client)
+      ->asset(id => 922756, asset => $a, type => 'hdd', host => $host);
+    $req->enqueue;
+    my $worker = $app->minion->repair->worker->register;
+    $worker->dequeue(0)->execute;
+    ok $req->processed;
+    ok $req->output;
+    like $req->output, qr/Initialized with localhost/;
+    ok $cache_client->asset_exists('localhost', $a);
+};
+
+subtest 'Test Minion Sync task' => sub {
+    my $a = 'sle-12-SP3-x86_64-0368-200_133333@64bit.qcow2';
+
+    use Minion;
+    use OpenQA::Worker::Cache::Task::Sync;
+    my $minion = Minion->new(SQLite => "sqlite:" . OpenQA::Worker::Cache->from_worker->db_file);
+    use Mojolicious;
+    my $app = Mojolicious->new;
+    $app->attr(minion => sub { $minion });
```

Comment:
You're changing the class `Mojolicious` here by adding a new attribute. Using a helper would be much safer (`$app->helper(minion => sub { $minion })`).

Reply:
You're creating the method `Mojolicious::minion` with `attr`. All future instances of `Mojolicious` in that test file will also have that method, and it will have higher precedence than a `minion` helper. Which could quickly get rather confusing if it interferes with a later test.

Reply:
I see your point, will change it